### PR TITLE
webkitgtk,wpewebkit: Bump-up to version 2.44.0

### PR DESCRIPTION
**Pull request by psaavedra**
* wpewebkit: Bump-up to version 2.44.0 and also:
    
  * Delete recipes for version 2.40.5
  * Update test jobs definitions to primary build 2.44.X
  * Remove preset for 2.40 and add a new one for 2.44
    
  * Release notes: https://wpewebkit.org/release/wpewebkit-2.44.0.html

* cog: Remove recipe for version 0.16.1
* libwpe: Add recipe for release 1.16.0
    
  * Release notes: https://wpewebkit.org/release/libwpe-1.16.0.html

* webkitgtk: Bump-up to version 2.44.0
    
  * Release notes: https://webkitgtk.org/2024/03/16/webkitgtk2.44.0-released.html

### CODE REVIEW

File: recipes-browser/cog/cog-meson.inc [hunk 1]
```diff
@@ -4,7 +4,7 @@ PACKAGECONFIG[dbus] = "-Dcog_dbus_control=system -Dcog_dbus_system_owner=${COG_D
 PACKAGECONFIG[drm] = ",,wpebackend-fdo libdrm virtual/libgbm libinput"
 PACKAGECONFIG[gtk4] = ",,gtk4"
 PACKAGECONFIG[headless] = ",,wpebackend-fdo"
-PACKAGECONFIG[soup2] = "-Dsoup2=enabled,-Dsoup2=disabled,libsoup-2.4"
+PACKAGECONFIG[soup2] = "-Dwpe_api=1.0,,libsoup-2.4"
```
Comment (aperezdc) [inline recipes-browser/cog/cog-meson.inc:7]: Isn't there `libsoup3` yet in Yocto? I would like to encourage switching to it if possible 😺

**Review by aperezdc — CHANGES_REQUESTED** [review]
The GTK port uses DMA-BUF now instead of the WPE renderer to achieve zero-copy passing graphics buffers from one process to another. Please remove the `PACKAGECONFIG[wperenderer]` entry from the `.bb` file.

File: recipes-browser/cog/cog-meson.inc (hunk 1 above)
Comment (psaavedra) [reply recipes-browser/cog/cog-meson.inc:7]: This is branch is still offering support to the dunfell and other intermediate Yocto release. libsoup 3 will be available from kirkstone and so. This is being addressed in https://github.com/Igalia/meta-webkit/pull/489

**Comment by psaavedra** [comment]
> wperenderer

Great. I will make the change.

**Review by aperezdc — APPROVED** [review]
Changes LGTM, thanks for removing the `wperenderer` option.

**Comment by psaavedra** [comment]
I think we will need to drop the support for old release before bumping up this wpewebkit version. The inability to use Yocto releases before Kirkstone stems from the fact that these older releases do not provide a compiler compatible with C++20, leading to compilation errors such as:

1. Error: No member named 'bit_width' in namespace 'std'.
2. Error: 'operator delete' cannot take a dependent type as the first parameter.

These errors prevent the successful compilation of the codebase, hindering development and deployment processes. Upgrading to Kirkstone or later releases of Yocto, which offer compatible compilers with support for C++20 features. This is necessary to address these issues right now.

**Comment by psaavedra** [comment]
Closed in favour of https://github.com/Igalia/meta-webkit/pull/496

**Pull request closed without merging**